packages/web-client/src/pages/profile/[username]/Profile.tsx

```diff
@@ -50,14 +87,48 @@ const ProfilePage: NextPage<Props> = ({ user }: Props) => {
             <p className="text-xl text-black-light">{user.bio}</p>
           </div>
         </div>
+        <ContentTabs className="py-4" tabs={['Courses', 'Videos', 'Blogs', 'Enrolled Courses', 'Bookmarks']}>
+          <Tab enableMore={courses.data.length < courses.count} loadMore={coursesLoadMore} title="Courses">
+            <div className="grid grid-flow-row md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-8 py-4">
+              <CoursesList courses={courses.data} />
+            </div>
+          </Tab>
+          <Tab enableMore={videos.data.length < videos.count} loadMore={videosLoadMore} title="Videos">
+            <div className="grid grid-flow-row md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-8 py-4">
+              <VideosList videos={videos.data} />
+            </div>
+          </Tab>
+          <Tab enableMore={blogs.data.length < blogs.count} loadMore={blogsLoadMore} title="Blogs">
+            <div className="grid grid-flow-row md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-8 py-4">
+              <BlogsList blogs={blogs.data} />
+            </div>
+          </Tab>
+          <Tab
+            enableMore={enrolledCourses.data.length < enrolledCourses.count}
+            loadMore={enrolledCoursesLoadMore}
+            title="Enrolled Courses"
+          >
+            <div className="grid grid-flow-row md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-8 py-4">
+              <CoursesList enrolled courses={courses.data} />
+            </div>
+          </Tab>
+          <Tab title="Bookmarks">
+            <InDevelopment inDevelopmentItem="feature" />
+          </Tab>
+        </ContentTabs>
       </div>
     </Layout>
   );
 };
 
 ProfilePage.getInitialProps = async (ctx) => {
   const user = await UsersService.getProfileInfo(ctx.query.username.toString());
-  return { user };
+  const initialBlogs = await UsersService.getBlogs(ctx.query.username.toString());
+  const initialVideos = await UsersService.getVideos(ctx.query.username.toString());
+  const initialCourses = await UsersService.getCourses(ctx.query.username.toString());
+  const initialEnrolledCourses = await UsersService.getEnrolledCourses(ctx.query.username.toString());
+
+  return { user, initialBlogs, initialVideos, initialCourses, initialEnrolledCourses };
```

Comment:
We should be using Promise.all here or the code will be too slow due to waiting for each call in a synchronous way.
```suggestion
  const [user, initialBlogs, initialVideos, initialCourses, initialEnrolledCourses] = await Promise.all([
    UsersService.getProfileInfo(ctx.query.username.toString()),
    UsersService.getBlogs(ctx.query.username.toString()),
    UsersService.getVideos(ctx.query.username.toString()),
    UsersService.getCourses(ctx.query.username.toString()),
    UsersService.getEnrolledCourses(ctx.query.username.toString()),
  ]);

  return { user, initialBlogs, initialVideos, initialCourses, initialEnrolledCourses };
```